javascript/modules/client/src/client-erc20.ts

```diff
@@ -434,21 +432,140 @@ export class ClientErc20 extends ClientCore implements IClientErc20 {
+  private async _getProposal(
+    proposalId: string,
+  ): Promise<Erc20Proposal | null> {
     if (!proposalId) {
       throw new Error("Invalid proposalId");
     }
-    const proposal = getDummyErc20Proposal(proposalId);
-    proposal.status = getProposalStatus(
-      proposal.startDate,
-      proposal.endDate,
-      true,
-      proposal.result.yes,
-      proposal.result.no,
-    );
-    return new Promise((resolve) => setTimeout(resolve, 1000)).then(
-      () => (proposal),
-    );
+    try {
+      await this.graphql.ensureOnline();
+      const client = this.graphql.getClient();
+      const { erc20VotingProposal } = await client.request(QueryErc20Proposal, {
+        proposalId,
+      });
+      if (!erc20VotingProposal) {
+        return null;
+      }
+      // TODO
+      // delete this cid once the proposals in subgraph have the correct
+      // format in the metadata field
+      const test_cid = "QmXhJawTJ3PkoKMyF3a4D89zybAHjpcGivkb7F1NkHAjpo";
+      const metadataString = await this.ipfs.fetchString(test_cid);
+      const metadata = JSON.parse(metadataString) as ProposalMetadata;
+      const startDate = new Date(
+        parseInt(erc20VotingProposal.startDate) * 1000,
+      );
+      const endDate = new Date(parseInt(erc20VotingProposal.endDate) * 1000);
+      const creationDate = new Date(
+        parseInt(erc20VotingProposal.createdAt) * 1000,
+      );
+      let usedVotingWeight: bigint = BigInt(0);
+      erc20VotingProposal.voters.forEach(
+        (voter: { voter: { id: string }; vote: string; weight: string }) => {
+          usedVotingWeight = usedVotingWeight + BigInt(voter.weight);
+        },
+      );
+      // get status
+      const status = getProposalStatus(
+        startDate,
+        endDate,
+        erc20VotingProposal.executed,
+        erc20VotingProposal
+          .yea,
+        erc20VotingProposal.nay,
+      );
+      // get start and end date
+      return {
+        id: erc20VotingProposal.id,
+        dao: {
+          address: erc20VotingProposal.dao.id,
+          name: erc20VotingProposal.dao.name,
+        },
+        creatorAddress: erc20VotingProposal.creator,
+        metadata: {
+          title: metadata.title,
+          summary: metadata.summary,
+          description: metadata.description,
+          resources: metadata.resources,
+          media: metadata.media,
+        },
+        startDate,
+        endDate,
+        creationDate,
+        actions: erc20VotingProposal.actions.map(
+          (action: { data: string; to: string; value: string }): DaoAction => {
+            return {
+              data: hexToBytes(strip0x(action.data)),
+              to: action.to,
+              value: BigInt(action.value),
+            };
+          },
+        ),
+        status,
+        result: {
+          yes: BigInt(erc20VotingProposal.yea),
+          no: BigInt(erc20VotingProposal.nay),
+          abstain: BigInt(erc20VotingProposal.abstain),
```

Comment:
Are we receiving subgraph data as numbers? or as strings? Big numbers will not be representable through a numeric graphql value

Reply:
`string` or `null` in some cases, here the null case is not taken into account, but in the next version it will be

Reply:
Strings